src/streamlink/stream/hls_playlist.py

```diff
@@ -238,9 +238,15 @@ def parse(self, data):
         self.state = {}
         self.m3u8 = M3U8()
 
-        for lineno, line in enumerate(filter(bool, data.splitlines())):
-            self.parse_line(lineno, line)
+        lines = enumerate(filter(bool, data.splitlines()))
+        _, line = next(lines)
+        if not line.startswith("#EXTM3U"):
+            raise ValueError("Missing #EXTM3U header")
 
+        parse_line = self.parse_line
+        for _, line in lines:
+            parse_line(line)
```

Comment:
why not `self.parse_line` as before

Reply:
That is actually a slight optimisation as member look ups are slower - however, I'm not convinced it makes any measurable difference in this case.